src/app/server/Server.cpp

```diff
@@ -762,23 +763,22 @@ void Server::ResumeSubscriptions()
 
 Credentials::IgnoreCertificateValidityPeriodPolicy Server::sDefaultCertValidityPolicy;
 
-KvsPersistentStorageDelegate CommonCaseDeviceServerInitParams::sKvsPersistenStorageDelegate;
-PersistentStorageOperationalKeystore CommonCaseDeviceServerInitParams::sPersistentStorageOperationalKeystore;
-Credentials::PersistentStorageOpCertStore CommonCaseDeviceServerInitParams::sPersistentStorageOpCertStore;
-Credentials::GroupDataProviderImpl CommonCaseDeviceServerInitParams::sGroupDataProvider;
-app::DefaultTimerDelegate CommonCaseDeviceServerInitParams::sTimerDelegate;
-app::reporting::ReportSchedulerImpl
-    CommonCaseDeviceServerInitParams::sReportScheduler(&CommonCaseDeviceServerInitParams::sTimerDelegate);
+Global<KvsPersistentStorageDelegate> CommonCaseDeviceServerInitParams::sKvsPersistenStorageDelegate;
```

Comment:
Instead of all these separate globals, you can create a `DefaultCommonCaseDeviceServerInitParams` class inside `Server.cpp` (could be  a sub-class of `CommonCaseDeviceServerInitParams` I think) with an appropriate default constructor, and a single `Global` constructing that. Then reference that global in `InitializeStaticResourcesBeforeServerInit` (which should probably move into the .cpp file) instead of those individual globals.

This also avoids having to add an `Init` method to `ReportScheduler`.

(A simpler existing example of this pattern is `struct ControllerAccessControl`.)

Reply:
Thinking about this more, it may be worthwhile to look at some of those individual default implementation objects and see how large they are, to determine if it makes sense to keep them as separate Globals. However the pattern of wrapping or sub-classing the ReportScheduler should still be used to avoid having to change the initialization.